dll/win32/browseui/utility.cpp

```diff
@@ -14,3 +17,57 @@ void operator delete(void *p, UINT_PTR)
 {
     LocalFree(p);
 }
+
+HRESULT SHELL_GetIDListFromObject(IUnknown *punk, PIDLIST_ABSOLUTE *ppidl)
+{
+#if DLL_EXPORT_VERSION >= _WIN32_WINNT_VISTA && 0 // FIXME: SHELL32 not ready yet
+    return SHGetIDListFromObject(punk, ppidl);
+#else
+    HRESULT hr;
+    IPersistFolder2 *pf2;
+    if (SUCCEEDED(hr = punk->QueryInterface(IID_PPV_ARG(IPersistFolder2, &pf2))))
+    {
+        hr = pf2->GetCurFolder(ppidl);
+        pf2->Release();
+    }
+    IPersistIDList *pil;
+    if (FAILED(hr) && SUCCEEDED(hr = punk->QueryInterface(IID_PPV_ARG(IPersistIDList, &pil))))
+    {
+        hr = pil->GetIDList(ppidl);
+        pil->Release();
+    }
+    return hr;
+#endif
+}
+
+static HRESULT SHELL_CompareAbsoluteIDs(LPARAM lParam, PCIDLIST_ABSOLUTE a, PCIDLIST_ABSOLUTE b)
+{
+    IShellFolder *psf;
+    HRESULT hr = SHGetDesktopFolder(&psf);
+    if (FAILED(hr))
+        return hr;
+    hr = psf->CompareIDs(lParam, a, b);
+    psf->Release();
+    return hr;
+}
+
+BOOL SHELL_IsEqualAbsoluteID(PCIDLIST_ABSOLUTE a, PCIDLIST_ABSOLUTE b)
+{
+    return 0 == SHELL_CompareAbsoluteIDs(SHCIDS_CANONICALONLY, a, b);
```

Comment:
```suggestion
    return (SHELL_CompareAbsoluteIDs(SHCIDS_CANONICALONLY, a, b) == 0);
```
Avoiding the yoda-style programming.

Reply:
What about uing
`return !SHELL_CompareAbsoluteIDs(SHCIDS_CANONICALONLY, a, b);`
instead of the suggested
`return (SHELL_CompareAbsoluteIDs(SHCIDS_CANONICALONLY, a, b) == 0);`

Wouldn't that work as well?

Reply:
It should. It's up to whindsaks.

Reply:
I prefer ! but I sometimes get nagged on it when I do it for Reg functions and you guys want == ERROR_SUCCESS.

Reply:
In this case the 2 of us preferring the `!` seem to be the majority already.
I am (same as you) no fan of nested braces. They do dramatically increase the mental load when reading code.
Therefore it is good to avoid them, if it is possible. And here it is possible.